app/posts/finishing-touches-for-forms.md

```diff
@@ -0,0 +1,103 @@
+---
+title: Finishing touches for forms
+date: 2025-01-05
+description: Some tips on making sure forms written in Ruby on Rails are accessible
+author:
+  name: Peter Yates
+  url: https://github.com/peteryates
+---
+
+The [GOV.UK formbuilder](https://govuk-form-builder.netlify.app) makes it easy for Ruby on Rails developers to build forms that follow the rules set out in the [GOV.UK design system](https://design-system.service.gov.uk/).
+
+It takes care of the markup, provides a nice API for customising the fields to meet the needs of your service and takes care of clearly dispalying any error messages.
+
+There are, however, some suggestions in the design system guidance that go beyond the remit of the form builder.
+
+They are often forgotten, we'll go over some simple ways to solve them here.
+
+## Positioning the error summary at the top of the page
+
+In addition to making it clear that something is wrong, we also need to tell the user exactly what happened and how to fix it.
+
+[The guidance says](https://design-system.service.gov.uk/components/error-summary#where-to-put-the-error-summary):
+
+> Put the error summary at the top of the main container. If your page includes breadcrumbs or a back link, place it below these, but above the `<h1>`.
+
+This causes a few problems.
+
+The GOV.UK design system includes some complex rules around linking to fields from the error summary. For example, errors on checkbox or radio button fields should link to the first checkbox or radio button.
```

Comment:
This bit isn’t explained in the post. Worth adding a separate section to explain `link_errors: true`?

Reply:
Yeah, this post needs three things; two seems too few!

Reply:
Header image too! 😀